File: 프로그래머스/0/181838. 날짜 비교하기/날짜 비교하기.cs
using System;

public class Solution 
{
    public int solution(int[] date1, int[] date2) 
    {
        int i = 0;
        bool isYear = false;
        bool isMonth = false;
        
        int answer = 0;
        if(date1[0] < date2[0])
        {
            answer = 1;
        }
        else if(date1[0] == date2[0])
        {
            isYear = true;
        }
        
        if(isYear && date1[1] < date2[1])
        {
            answer = 1;
        }
        else if(isYear && date1[1] == date2[1])
        {
            isMonth = true;
        }
        
        if(isYear && isMonth && date1[2] < date2[2])
        {
            answer = 1;
        }
        
        
        return answer;
    }
}

File: 프로그래머스/0/181838. 날짜 비교하기/날짜 비교하기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> date1, vector<int> date2) 
{
    int year1 = date1[0];
    int year2 = date2[0];
    
    int month1 = date1[1];
    int month2 = date2[1];
    
    int day1 = date1[2];
    int day2 = date2[2];
    
    if(year1 < year2)
    {
        return 1;
    }
    else if(year1 == year2)
    {
        if(month1 < month2)
        {
            return 1;
        }
        else if(month1 == month2)
        {
            if(day1 < day2)
            {
                return 1;
            } 
        }
    }
    return 0;
}